support-frontend/assets/pages/weekly-subscription-checkout/components/weeklyCheckoutForm.test.tsx

```diff
@@ -1,44 +1,70 @@
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+
 import '__mocks__/stripeMock';
-import { combineReducers, configureStore } from '@reduxjs/toolkit';
 import { fireEvent, screen } from '@testing-library/react';
 import { mockFetch } from '__mocks__/fetchMock';
 import { weeklyProducts } from '__mocks__/productInfoMocks';
 import { renderWithStore } from '__test-utils__/render';
-import { isSwitchOn } from 'helpers/globalsAndSwitches/globals';
 import { GuardianWeekly } from 'helpers/productPrice/subscriptions';
+import {
+	setBillingCountry,
+	setDeliveryCountry,
+} from 'helpers/redux/checkout/address/actions';
+import { setProductPrices } from 'helpers/redux/checkout/product/actions';
 import { setInitialCommonState } from 'helpers/redux/commonState/actions';
-import { commonReducer } from 'helpers/redux/commonState/reducer';
 import type { WithDeliveryCheckoutState } from 'helpers/subscriptionsForms/subscriptionCheckoutReducer';
-import { createReducer } from 'helpers/subscriptionsForms/subscriptionCheckoutReducer';
+import { setFirstName } from 'pages/contributions-landing/contributionsLandingActions';
+import { isSwitchOn } from '../../../helpers/globalsAndSwitches/globals';
+import type { BillingPeriod } from '../../../helpers/productPrice/billingPeriods';
+import type { FulfilmentOptions } from '../../../helpers/productPrice/fulfilmentOptions';
+import { getWeeklyFulfilmentOption } from '../../../helpers/productPrice/fulfilmentOptions';
+import type { ProductOptions } from '../../../helpers/productPrice/productOptions';
+import { NoProductOptions } from '../../../helpers/productPrice/productOptions';
+import type { SubscriptionsStore } from '../../../helpers/redux/subscriptionsStore';
+import { initReduxForSubscriptions } from '../../../helpers/redux/subscriptionsStore';
+import { formatMachineDate } from '../../../helpers/utilities/dateConversions';
 import WeeklyCheckoutForm from './weeklyCheckoutForm';
 
 function setUpStore(initialState: WithDeliveryCheckoutState) {
-	const store = configureStore({
-		reducer: combineReducers({
-			page: createReducer(),
-			common: commonReducer,
-		}),
-		preloadedState: initialState,
-	});
+	const store = initReduxForSubscriptions(
+		GuardianWeekly,
+		'Monthly',
+		formatMachineDate(new Date()),
+		NoProductOptions,
+		getWeeklyFulfilmentOption,
+	);
+	store.dispatch(setProductPrices(weeklyProducts));
 	store.dispatch(setInitialCommonState(initialState.common));
 	return store;
 }
 
-jest.mock('helpers/globalsAndSwitches/globals', () => ({
-	__esModule: true,
-	isSwitchOn: jest.fn(),
-	getSettings: jest.fn(),
-	getGlobal: jest.fn(),
-}));
+jest.mock('helpers/globalsAndSwitches/globals', () => {
+	const actualGlobalsAndSwitches = jest.requireActual(
+		'helpers/globalsAndSwitches/globals',
+	);
 
+	return {
+		...actualGlobalsAndSwitches,
+		isSwitchOn: jest.fn(),
+	};
+});
```

Comment:
Just wondering if it's worth pulling this out into a separate file like the Stripe mock, to make it reusable.

Reply:
Good shout! I'm off tomorrow, but could do this on Monday. Probably makes sense to do it in a separate PR though, so this can get merged sooner rather than later.

Reply:
Yeah that makes sense 👍